src/base/NekRSProblemBase.C

```diff
@@ -629,23 +636,39 @@ NekRSProblemBase::externalSolve()
   if (_t_step <= 1000)
     nekrs::verboseInfo(true);
 
-  // Tell NekRS what the time step size is
-  nekrs::initStep(_timestepper->nondimensionalDT(step_start_time),
-                  _timestepper->nondimensionalDT(_dt),
-                  _t_step);
-
   // Run a nekRS time step. After the time step, this also calls UDF_ExecuteStep,
   // evaluated at (step_end_time, _t_step) == (nek_step_start_time + nek_dt, t_step)
-  int corrector = 1;
-  bool converged = false;
-  do
+  if (_fp_iteration)
   {
-    converged = nekrs::runStep(corrector++);
-  } while (!converged);
-
-  // TODO: time is somehow corrected here
-  nekrs::finishStep();
+    const PostprocessorValue * iter = &getPostprocessorValueByName("fp_iteration");
+    std::cout << "Current fixed point iteration number read in NekRSProblemBase is " << *iter
+              << std::endl; // JUST FOR TESTING
```

Comment:
Is this being kept in the final code? Comment indicates otherwise

Reply:
I think this should be kept in the final code. Depending on how the postprocessor for the iteration is passed into nek.i, it can be possible to mess it up (if the correct execute_on flags are not used), so I think this is necessary for the user to check that the correct iteration count is being received.

Reply:
Maybe the comment should be removed, otherwise everything looks good to me, thanks @regerdavid @aprilnovak